### PR TITLE
[T1007-MG < T1014-MG] Add authorization to read operators

### PR DESCRIPTION
**Pull request by Josipmrden**
[T1007 < T1014] Add authorization to read operators PR
- [X] Write functionality
- [X] Write E2E tests

### CODE REVIEW

File: tests/e2e/lba_procedures/read_permission_queries.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+import pytest
```
Comment (BorisTasevski) [inline tests/e2e/lba_procedures/read_permission_queries.py:13]: sort imports alphabetically

File: tests/e2e/lba_procedures/read_permission_queries.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+def get_user_cursor():
+    return connect(username="user", password="test").cursor()
+
+
+def test_can_read_node_without_index():
+    admin_cursor = get_admin_cursor()
+    user_cursor = get_user_cursor()
+
+    grant_groups = [
+        ["GRANT READ ON LABELS :read_label TO user;"],
```
Comment (BorisTasevski) [inline tests/e2e/lba_procedures/read_permission_queries.py:38]: Grant groups could have been defined on module level similar to `match_by_label_query`

File: tests/e2e/lba_procedures/read_permission_queries.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            results = execute_and_fetch_all(user_cursor, mq)
+            assert len(results) == 1
+
+
+def test_can_read_node_with_index():
+    admin_cursor = get_admin_cursor()
+    user_cursor = get_user_cursor()
+
+    grant_groups = [
+        ["GRANT READ ON LABELS :read_label TO user;"],
```
Comment (BorisTasevski) [inline tests/e2e/lba_procedures/read_permission_queries.py:62]: Also, the code is duplicated

File: tests/e2e/lba_procedures/read_permission_queries.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        match_by_label_property_range_query,
+        match_by_label_property_value_query,
+    ]
+
+    for grant_group in grant_groups:
+        reset_permissions(admin_cursor)
+        execute_and_fetch_all(admin_cursor, "CREATE INDEX ON :read_label;")
+        execute_and_fetch_all(admin_cursor, "CREATE INDEX ON :read_label(prop);")
+
+        [execute_and_fetch_all(admin_cursor, x) for x in grant_group]
```
Comment (BorisTasevski) [inline tests/e2e/lba_procedures/read_permission_queries.py:163]: What are we doing with this list? Use simple for loop, the intent is more clear that way. Here some phantom array is created and then not used

**Review by BorisTasevski — CHANGES_REQUESTED** [review]

File: tests/e2e/lba_procedures/read_permission_queries.py (hunk 1 above)
Comment (Josipmrden) [reply tests/e2e/lba_procedures/read_permission_queries.py:12]: Done

File: tests/e2e/lba_procedures/read_permission_queries.py (hunk 2 above)
Comment (Josipmrden) [reply tests/e2e/lba_procedures/read_permission_queries.py:30]: Done, put in methods

File: tests/e2e/lba_procedures/read_permission_queries.py (hunk 3 above)
Comment (Josipmrden) [reply tests/e2e/lba_procedures/read_permission_queries.py:41]: Done, added some grupation

**Review by BorisTasevski — APPROVED** [review]

File: tests/e2e/lba_procedures/read_permission_queries.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+match_query = "MATCH (n) RETURN n;"
+match_by_id_query = "MATCH (n) WHERE ID(n) >= 0 RETURN n;"
+
+match_by_label_query = "MATCH (n:read_label) RETURN n;"
+match_by_label_property_range_query = "MATCH (n:read_label) WHERE n.prop < 7 RETURN n;"
+match_by_label_property_value_query = "MATCH (n:read_label {prop: 5}) RETURN n;"
+match_by_label_property_query = "MATCH (n:read_label) WHERE n.prop IS NOT NULL RETURN n;"
+
+
+def get_read_node_without_index_operation_cases() -> List[str]:
```
Comment (jbajic) [inline tests/e2e/lba_procedures/read_permission_queries.py:28]:
```suggestion
def get_read_node_without_index_operation_cases() -> List[List[str]]:
```

Comment (jbajic) [reply tests/e2e/lba_procedures/read_permission_queries.py:28]: And why not make these just variables, that you can use in the test immediately

File: tests/e2e/lba_procedures/workloads.yaml [hunk 6]
```diff
@@ -3,21 +3,36 @@ template_cluster: &template_cluster
...
+          "Create (:Label1 {id: 5}) ;",
+          "Create (:Label1 {id: 6}) ;",
+          "Create (:Label2 {id: 1}) ;",
+          "Create (:Label2 {id: 2}) ;",
+          "Create (:Label2 {id: 3}) ;",
+          "Create (:Label2 {id: 4}) ;",
+          "Create User Josip ;",
+          "Create User Boris ;",
+          "Grant Read On Labels :Label1 to Boris;",
+        ]
```
Comment (jbajic) [inline tests/e2e/lba_procedures/workloads.yaml:21]:
```suggestion
      setup_queries:
          - "Create (:Label1 {id: 1}) ;"
          - "Create (:Label1 {id: 2}) ;"
          - "Create (:Label1 {id: 3}) ;"
          - "Create (:Label1 {id: 4}) ;"
          - "Create (:Label1 {id: 5}) ;"
          - "Create (:Label1 {id: 6}) ;"
          - "Create (:Label2 {id: 1}) ;"
          - "Create (:Label2 {id: 2}) ;"
          - "Create (:Label2 {id: 3}) ;"
          - "Create (:Label2 {id: 4}) ;"
          - "Create User Josip ;"
          - "Create User Boris ;"
          - "Grant Read On Labels :Label1 to Boris;"
```

File: tests/e2e/lba_procedures/workloads.yaml [hunk 7]
```diff
@@ -3,21 +3,36 @@ template_cluster: &template_cluster
...
+    main:
+      args: ["--bolt-port", "7687", "--log-level=TRACE"]
+      log_file: "lba-e2e.log"
+      setup_queries:
+        [
+          "CREATE USER admin IDENTIFIED BY 'test';",
+          "GRANT ALL PRIVILEGES TO admin;",
+          "CREATE USER user IDENTIFIED BY 'test';",
+          "GRANT ALL PRIVILEGES TO user;",
+        ]
```
Comment (jbajic) [inline tests/e2e/lba_procedures/workloads.yaml:35]: Same as above, `[]` should only be used for the one-line list in YAML (https://yaml.org/spec/1.2.2/#21-collections)

File: tests/e2e/lba_procedures/read_permission_queries.py (hunk 5 above)
Comment (Josipmrden) [reply tests/e2e/lba_procedures/read_permission_queries.py:28]: Okay, makes sense, will make them into global variables

**Review by antaljanosbenjamin — APPROVED** [review]

**Review by jbajic — APPROVED** [review]

**Pull request merged**